core/evm.go

```diff
@@ -64,6 +69,7 @@ func NewEVMBlockContext(header *types.Header, chain ChainContext, author *common
 		Time:          header.Time,
 		Difficulty:    new(big.Int).Set(header.Difficulty),
 		BaseFee:       baseFee,
+		BlobFee:       blobFee,
```

Comment:
There are a lot of places that doesn't use this constructor. (Maybe we ought to make them use a constructor, becuause it's very easy to miss updating one of them)
![Screenshot from 2023-09-14 09-51-50](https://github.com/ethereum/go-ethereum/assets/142290/f582502a-1107-4520-a585-20597f8cdfc2)

`46` places where we create a `vm.BlockContext` struct

Reply:
The problem is that most of these cases are in tests where we only set one or two of the fields. We can't really update them to use the constructor since that will need a header to populate the fields. We could theoretically create a constructor that will set every field, but that will result in a lot of `NewBlockContext(nil, nil, .-.. nil)`. Also we pass the struct by value very often so most of the initializatios are actually `BlockCOntext{}`